cranelift/codegen/src/machinst/compile.rs

```diff
@@ -0,0 +1,63 @@
+//! Compilation backend pipeline: optimized IR to VCode / binemit.
+
+use crate::ir::Function;
+use crate::machinst::*;
+use crate::settings;
+use crate::timing;
+
+use log::debug;
+use regalloc::{allocate_registers, RegAllocAlgorithm};
+
+/// Compile the given function down to VCode with allocated registers, ready
+/// for binary emission.
+pub fn compile<B: LowerBackend>(
+    f: &mut Function,
```

Comment:
This can take &Function I think.

Reply:
Yup, this was left over from an earlier version that did mutate the function; much nicer now to take an immutable input. Thanks!